verification/cbmc/include/proof_helpers/make_common_data_structures.h

```diff
@@ -19,8 +19,14 @@
 
 #include <stdlib.h>
 
-#define ASSUME_VALID_MEMORY(ptr) ptr = bounded_malloc(sizeof(*(ptr)))
-#define ASSUME_VALID_MEMORY_COUNT(ptr, count) ptr = bounded_malloc(sizeof(*(ptr)) * (count))
+#define ASSUME_VALID_MEMORY_COUNT(ptr, count)                                                                          \
+    do {                                                                                                               \
+        ptr = bounded_malloc(sizeof(*(ptr)) * (count));                                                                \
```

Comment:
I'm not sure we need bounded malloc at all here. CBMC should already handle returning `null` if you ask for too much memory

Reply:
We still need bounded_malloc not only to bound the input but also to cover the case when the input is 0 and malloc return a invalid input.

Reply:
We should decide what to do in this case, but I won't block the PR on it.